core/src/main/java/jenkins/telemetry/impl/UserLanguages.java

```diff
@@ -0,0 +1,135 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.telemetry.impl;
+
+import hudson.Extension;
+import hudson.init.Initializer;
+import hudson.util.PluginServletFilter;
+import jenkins.telemetry.Telemetry;
+import net.sf.json.JSONObject;
+import org.apache.commons.lang.mutable.MutableLong;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import javax.annotation.Nonnull;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Extension
+@Restricted(NoExternalUse.class)
+public class UserLanguages extends Telemetry {
+
+    private static Map<String, MutableLong> requestsByLanguage = new TreeMap<>();
+    private static Logger LOGGER = Logger.getLogger(UserLanguages.class.getName());
+
+    @Nonnull
+    @Override
+    public String getId() {
+        return UserLanguages.class.getName();
+    }
+
+    @Nonnull
+    @Override
+    public String getDisplayName() {
+        return "Browser languages";
+    }
+
+    @Nonnull
+    @Override
+    public LocalDate getStart() {
+        return LocalDate.of(2018, 10, 1);
+    }
+
+    @Nonnull
+    @Override
+    public LocalDate getEnd() {
+        return LocalDate.of(2019, 1, 1);
+    }
+
+    @Nonnull
+    @Override
+    public JSONObject createContent() {
+        JSONObject payload = new JSONObject();
+        for (Map.Entry<String, MutableLong> entry : requestsByLanguage.entrySet()) {
+            payload.put(entry.getKey(), entry.getValue().longValue());
+        }
+        return payload;
+    }
+
+    @Initializer
+    public static void setUpFilter() {
+        Filter filter = new AcceptLanguageFilter();
+        if (!PluginServletFilter.hasFilter(filter)) {
+            try {
+                PluginServletFilter.addFilter(filter);
+            } catch (ServletException ex) {
+                LOGGER.log(Level.WARNING, "Failed to set up languages servlet filter", ex);
+            }
+        }
+    }
+
+    public static final class AcceptLanguageFilter implements Filter {
+
+        @Override
+        public void init(FilterConfig filterConfig) {
+
+        }
+
+        @Override
+        public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
```

Comment:
Would there be any way make this unique based on cookie or other browser/session-specific identifier? Otherwise, if I am understanding this code correctly, a single power-user would skew the statistics quite a bit.

Reply:
Multiple sessions over a period make this essentially impossible, and I'm not comfortable with further browser fingerprinting. Note that we know # of requests by instance ID, which might allow us to disregard spikes.

Reply:
Okie doke